include/boost/unordered/detail/foa/core.hpp

```diff
@@ -1350,6 +1407,19 @@ table_core:empty_value<Hash,0>,empty_value<Pred,1>,empty_value<Allocator,2>
   using locator=table_locator<group_type,element_type>;
   using arrays_holder_type=arrays_holder<arrays_type,Allocator>;
 
+protected:
+  /* These traits are used in function implementations */
+  using is_map =
+    std::integral_constant<bool, !std::is_same<key_type, value_type>::value>;
+  using is_key_type_move_constructible =
+    typename std::conditional<detail::is_complete<key_type>::value,
+      std::is_move_constructible<key_type>, std::false_type>::type;
+  template <typename K>
+  using is_emplace_kv_able = std::integral_constant<bool,
+    is_map::value && (detail::is_similar<K, key_type>::value ||
+                       is_key_type_move_constructible::value)>;
+
```

Comment:
Is there a way to keep these traits from having to be protected member typedefs?

Reply:
They can be free traits instead. I'm not sure about the naming though. I'll take any suggestions